apps/condo/domains/marketplace/schema/MarketItem.test.js

```diff
@@ -0,0 +1,307 @@
+            })
+        })
+
+        describe('user staff without access', () => {
+            let organization, client
+            beforeAll(async () => {
+                [organization] = await createTestOrganization(admin)
+                client = await makeClientWithNewRegisteredAndLoggedInUser()
+                const [role] = await createTestOrganizationEmployeeRole(admin, organization, {
+                    canReadMarketItems: false,
+                    canManageMarketItems: false,
+                })
+                await createTestOrganizationEmployee(admin, organization, client.user, role)
+            })
+
+            test('can\'t create', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await createTestMarketItem(client, marketCategory, organization)
+                })
+            })
+
+            test('can\'t update', async () => {
+                const [objCreated] = await createTestMarketItem(admin, marketCategory, organization)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await updateTestMarketItem(client, objCreated.id)
+                })
+            })
+
+            test('can\'t delete', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await MarketItem.delete(client, 'id')
+                })
+            })
+
+            test('can\'t read', async () => {
+                await createTestMarketItem(admin, marketCategory, organization)
+
+                const objs = await MarketItem.getAll(client, {}, { sortBy: ['updatedAt_DESC'] })
+
+                expect(objs).toHaveLength(0)
+            })
+        })
+
+        describe('anonymus', () => {
+            let organization, client
+            beforeAll(async () => {
+                [organization] = await createTestOrganization(admin)
+                client = await makeClient()
+            })
+            test('can\'t create', async () => {
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await createTestMarketItem(client, marketCategory, organization)
+                })
+            })
+
+            test('can\'t update', async () => {
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await updateTestMarketItem(client, 'id')
+                })
+            })
+
+            test('can\'t delete', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await MarketItem.delete(client, 'id')
+                })
+            })
+
+            test('can\'t read', async () => {
+                await expectToThrowAuthenticationErrorToObjects(async () => {
+                    await MarketItem.getAll(client, {}, { sortBy: ['updatedAt_DESC'] })
+                })
+            })
+        })
+    })
+
+    describe('resident', () => {
```

Comment:
Why is block with tests for resident accesses outside block with other accesses?